src/components/Button/Button.tsx

```diff
@@ -22,6 +22,7 @@ export const Button: React.FC<ButtonProps> = ({
   type,
   className,
   isDisabled = false,
+  ...rest
```

Comment:
We're adding this because we are now extending from an existing type correct?

I can see some potential issues with a pattern like this.  It's commonly argued as React anti-pattern.  This is due to the lack of clarity and the potential for garbage props that result in unnecessary re-renders.

We should think carefully before adding this pattern to the code base. 

Is it worth it or necessary?  Does everyone know the potential downsides and issues that can arise from it?  Do we have any maintainability concerns?  Where can we add it and where should be not? Etc..

I recommend doing some research on this pattern.

My opinion is that it's not worth the hassle -- but one can be convinced.

Reply:
Yeah I think the point is to extend the native input type here so we don't have to explicitly define every single `HTMLButtonElement` type. Is there a better way to do this?

Reply:
Good point!  I'll look into this a bit more.  For now I have removed all of the additions in this PR.
Can you share any articles you have come across discussing re-rendering issues?  The stuff I have read recently suggests that extending customized inputs is the way to go.

Reply:
After doing some testing, I appears that we are missing one or more attributes that are required by React-Hook-Form to do its work.  Would either of you happen to know off the top of your head which one(s) would be required?

Reply:
Hey @Thistle-Strum, let's go ahead and test out using `...rest`, but rename it to be more specific.  Perhaps `...baseElementProps`.

I did a bit more digging into this and it's not as bad since we are using TypeScript.